omnigibson/robots/franka.py

```diff
@@ -84,9 +86,96 @@ def __init__(
                 If "physical", no assistive grasping will be applied (relies on contact friction + finger force).
                 If "assisted", will magnetize any object touching and within the gripper's fingers.
                 If "sticky", will magnetize any object touching the gripper's fingers.
+            end_effector (str): type of end effector to use. One of {"gripper", "allegro", "leap_right", "leap_left", "inspire"}
             kwargs (dict): Additional keyword arguments that are used for other super() calls from subclasses, allowing
                 for flexible compositions of various object subclasses (e.g.: Robot is USDObject + ControllableObject).
         """
+        # store end effector information
+        self.end_effector = end_effector
+        if end_effector == "gripper":
+            self._model_name = "franka_panda"
+            self._gripper_control_idx = np.arange(7, 9)
+            self._eef_link_names = "panda_hand"
+            self._finger_link_names = ["panda_leftfinger", "panda_rightfinger"]
+            self._finger_joint_names = ["panda_finger_joint1", "panda_finger_joint2"]
+            self.__default_joint_pos = np.array([0.00, -1.3, 0.00, -2.87, 0.00, 2.00, 0.75, 0.00, 0.00])
+            self._teleop_rotation_offset = np.array([-1, 0, 0, 0])
+            self._ag_start_points = [
+                GraspingPoint(link_name="panda_rightfinger", position=[0.0, 0.001, 0.045]),
+            ]
+            self._ag_end_points = [
+                GraspingPoint(link_name="panda_leftfinger", position=[0.0, 0.001, 0.045]),
+            ]
+        elif end_effector == "allegro":
+            self._model_name = "franka_allegro"
+            # thumb.proximal, ..., thumb.tip, ..., ring.tip
+            self._gripper_control_idx = np.array([8, 12, 16, 20, 10, 14, 18, 22, 9, 13, 17, 21, 7, 11, 15, 19])
```

Comment:
what is this exactly?

Reply:
It's the control index of the gripper, basically we set it this way so that for example if we use JointController for dexhand we will be controlling thumb, then index finger, then middle finger... in order.